vernac/comHints.ml

```diff
@@ -64,11 +64,11 @@ let project_hint ~poly pri l2r r =
   (info, false, true, Hints.PathAny, Hints.IsGlobRef (Names.GlobRef.ConstRef c))
 
 let warn_deprecated_hint_constr =
-  CWarnings.create ~name:"deprecated-hint-constr" ~category:"deprecated"
+  CWarnings.create ~name:"fragile-hint-constr" ~category:"automation"
     (fun () ->
       Pp.strbrk
-        "Declaring arbitrary terms as hints is deprecated; declare a global \
-         reference instead")
+        "Declaring arbitrary terms as hints is fragile; it is recommended to \
+         declare a global reference instead")
```

Comment:
The term "global reference" may be too technical in this context?

Reply:
What would you prefer to have there?

Reply:
No concrete idea, as discussed on the 8.12 release thread on Zulip I found the warning a bit confusing for users; in the sense that if I'm a user what I'm supposed to do here; why is this fragile? Am I in danger or not? It would be a bit messy that we have users adding dozens of new dummy definitions just to find out that we can support this better.

Maybe it would be good to add some documentation on the hint command itself.

Going back to the comment, I dunno, maybe "top-level constant" or "give the hint a name"; I'm sorry I don't have a better idea.

Reply:
I think "global reference" is OK as it is used sometimes in the documentation, but I do prefer the suggestion of "top-level constant".

Reply:
Done.

Reply:
Constructors are ok too though

Reply:
and variables (eg from pose)

Reply:
The warning will not trigger for those, so I think it's fine.